lib/linguist/generated.rb

```diff
@@ -58,6 +58,7 @@ def generated?
       node_modules? ||
       go_vendor? ||
       npm_shrinkwrap? ||
+      yarn_lock? ||
```

Comment:
Could you move this to the end of the list instead?
This list of functions starts with the functions that do not need to load the file (only use the extension and path).